### PR TITLE
CSpell update for POMs

### PR DESCRIPTION
**Pull request by JimSuplizio**
I noticed with my last PR that there were 90+ warnings coming out of the spell check and the vast majority of these were in POM files. What I did was to take a general look at POMs and made the changes to specifically target things like group, artifact, version tags, class names etc., that wouldn't, due to their natures, pass a spell check. The included README.md is going to have a far more detailed description than what I'm going to give here.

My additions to cspell.json specifically target poms and what's in there right now. I'll conservatively say this PR fixes about 95% of the warnings we'd see from POM files but of the 200 or so I checked, there were less than 5 issues and those that I saw were because of aliases in comments.

@danieljurek, I had a brief chat with @weshaggard this morning and we both agreed that we shouldn't be scanning the eng directory. This tool should be more focused on directories that contain shipping objects and not engineering system pieces. Further, it was also decided to remove *.yml from the list as those aren't shipping. If we encounter a library that starts shipping yml, we'll have to change this but we'll cross that bridge when we come to it.

@saragluna The change to the spring pom was to remove a superfluous comment and nothing more.

### CODE REVIEW

**Comment by JimSuplizio** [comment]
The spring pipeline is failing due to missing javadocs and that already has a fix that was reviewed.

File: .vscode/README.md
Comment (danieljurek) [inline .vscode/README.md:37]: We do have an exception for this today. We want to validate the spelling that goes into the `[CHANGELOG.md](https://github.dev/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/CHANGELOG.md#L29)` files and sometimes those contain github aliases for contributors. In those cases we [add those aliases to `words`](https://github.dev/Azure/azure-sdk-for-cpp/blob/main/.vscode/cspell.json#L70) ... However, this is a good place to consider adding them to a specific override for `CHANGELOG.md` files so they don't, for example, get scooped up in later efforts to contribute to upstream dictionaries. 

https://github.com/Azure/azure-sdk-for-cpp/issues/3660

File: .vscode/README.md
Comment (danieljurek) [inline .vscode/README.md:9]: This section and the sections below it should be in https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/spellcheck.md so that other languages can benefit from the knowledge here. 

This file should provide the basic local testing instructions and link off to https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/spellcheck.md for the deeper dive into how spell checking works.

**Review by danieljurek — COMMENTED** [review]
This looks good overall. Just a quick refactoring of the docs so that everyone can benefit from the knowledge you've gained. 

This is one of the few examples where a regex is a good idea. Since the spell checker doesn't have a deeper semantic understanding of the files its checking this bridges the need to check human readable text in the `pom.xml` files while skipping over checking things that are machine-readable and not necessarily present in the configured dictionaries.

**Review by danieljurek — APPROVED** [review]
Approving so as not to block after the refactor is complete

**Review by weshaggard — COMMENTED** [review]
I defer to @danieljurek for signing off but I don't object to any of the changes here.

File: .vscode/README.md
Comment (JimSuplizio) [reply .vscode/README.md:38]: @danieljurek, Java doesn't put aliases, initials or names into changelogs. I created this set of rules based upon what I saw and, also, what made sense.

File: .vscode/README.md
Comment (JimSuplizio) [reply .vscode/README.md:10]: @danieljurek, I added the link to common doc in my readme. I'm going to leave it to you to update the common one with the rules, from this readme, that you think apply to all languages.

**Pull request merged**
